x-pack/plugin/downsample/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/ILMDownsampleDisruptionIT.java

```diff
@@ -172,6 +173,18 @@ public void testILMDownsampleRollingRestart() throws Exception {
         startDownsampleTaskViaIlm(sourceIndex, targetIndex);
         assertBusy(() -> assertTargetIndex(cluster, targetIndex, indexedDocs));
         ensureGreen(targetIndex);
+        // We assert that ILM to successfully completed the phase
+        assertBusy(() -> {
```

Comment:
Could we do this with `ClusterServiceUtils#addTemporaryStateListener` rather than polling any API? We can check the API response afterwards ofc but every `assertBusy` adds a little extra idle time to tests and in total that's a pretty big contributor to overall test runtime.